include/mbedtls/config_psa.h

```diff
@@ -921,6 +1038,56 @@ extern "C" {
 #define PSA_WANT_ALG_SOME_PAKE 1
 #endif
 
+/* Temporary internal migration helpers */
+#if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_USE) && \
```

Comment:
I don't understand the logic here: so if e.g. `USE` and `IMPORT` are requested but not `GENERATE` then we don't enable `LEGACY` and as a result PSA won't support ECC? Shouldn't this all be `||` instead of `&&` here, so that during the transition period we might implement _more_ than what is requested, not less?

We don't need to test all combinations, but I'd like to have at least one test that requests partial support, with validation that the requested support is present.

Reply:
Eventually I expect that we'll want to have good coverage. In fact I'd expect that the epic includes an issue for that @mpg? https://github.com/Mbed-TLS/mbedtls/issues/7673 is not really related: it's about getting correct dependencies in the test script, but we don't need that as long as the internal dependencies are still the all-encompassing `_LEGACY` symbols. But at least for now, I want a test that would have caught this error. That means an `all.sh` job (probably the right solution for now) or a configuration enumerated by `depends.py` (probably the right solution for better coverage later). That configuration should have something like `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_USE` and `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_IMPORT` explicitly enabled but `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_GENERATE` and `MBEDTLS_GENPRIME` not explicitly enabled, and check that the result builds correctly and supports doing an RSA signature with a fixed key (but we don't yet expect that it doesn't include the code for RSA key generation).

Reply:
>https://github.com/Mbed-TLS/mbedtls/issues/7673 is not really related: it's about getting correct dependencies in the test script, but we don't need that as long as the internal dependencies are still the all-encompassing _LEGACY symbols.

Are we reading the same 7673? Title "Replace PSA_WANT's LEGACY symbols for key pairs with proper ones (use, import, export, generate, derive)", and in the description "Now these _LEGACY symbols should be replaced with proper USE, IMPORT, EXPORT, GENERATE, DERIVE". Fixing the python script is only a side point as I read it.

> Eventually I expect that we'll want to have good coverage. In fact I'd expect that the epic includes an issue for that @mpg?

Good point, I forgot about that, sorry. I suggest we start in this PR with a minimal test, and think a bit more about what's needed next.

As discussed privately with Valerio, 7673 as its stands is probably a bit too big and we'll probably want to split it up. (Perhaps along the lines of : ECC, RSA, DH.) Depending on the testing strategy, perhaps each part will include its own testing, and/or we'll have a separate task for testing.

> That configuration should have something like `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_USE` and `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_IMPORT` explicitly enabled but `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_GENERATE` and `MBEDTLS_GENPRIME` not explicitly enabled, and check that the result builds correctly and supports doing an RSA signature with a fixed key (but we don't yet expect that it doesn't include the code for RSA key generation).

How do you propose we ensure the build supports doing RSA signatures? Grep the object file for the relevant function and asserting it's present? Grep the outcome file for a positive test case involving RSA signatures and asserting it passed, as opposed to being skipped?

(Note: in general I'm not sure we already have testing focused on ensuring that features that should be available in a given build actually are. It mostly relies on the fact that test dependencies are correct (specifically, not requiring more than needed), so the tests are run in builds where the feature is supposed to be present, and they would fail if it were not. But that's not really checked, if we got both the code and the dependency declarations wrong, then the tests would be skipped, the CI would pass and we could fail to notice other than by reviewing, as happened here.)

Reply:
> How do you propose we ensure the build supports doing RSA signatures?

We can check that an RSA test passes (relying on the outcome file). Or we can run one of the pk-based programs in `programs/pkey`.

> in general I'm not sure we already have testing focused on ensuring that features that should be available in a given build actually are

Indeed, we normally don't do that because we do gray box testing. We know that enabling `MBEDTLS_FOO` enables the tests that depend on `MBEDTLS_FOO`, unless we've made a mistake in _all_ of those tests. But here we have a (temporary) situation where tests are gated by one symbol (`xxx_LEGACY`) but the configuration defines other symbols (`xxx`, `xxx_USE`, etc.). So there is a potential for a mistake in the implication from those other symbols to the symbol used in the tests — and we did in fact make a mistake! So we should test for such a mistake.

Reply:
I found that the `component_test_psa_crypto_config_accel_rsa_signature` seem to do something similar to what we were trying to achieve here, so I added a new component which uses this old one, but defines new symbols instead of the legacy ones.

Reply:
I'm not sure it's a good idea to bring drivers into the mix. The primary objective is to check that if the application requests partial RSA support, it does get RSA support. Doing the testing via a driver also exercises the driver symbols to some extent (at least if they're wrong they have to be somewhat consistently wrong with the `PSA_WANT` symbols), but it also makes the test logic even harder to follow.

Reply:
I agree that we should keep the test simple and not involve drivers as that's not necessary, but let's grep the outcome file to ensure (at least) one positive test has indeed passed.

Reply:
I just updated the test (https://github.com/Mbed-TLS/mbedtls/pull/7641/commits/bd3235b37121345987b9f165758d4c0b911b48cd) so that:
- it does not uses drivers
- it checks for a specific output line of a test which performs RSA signature

In order to make this component quick, I just run the `test_suite_rsa` since it's the only one we're interested in. Is that reasonable to you?

Reply:
I would prefer to run all the tests. The tests are quick enough that it's no big deal to run the RSA test suite twice (but we can grep the outcome file to avoid that — however that means the script will fail if outcome files are disabled by running `all.sh --outcome-file=`).

Reply:
Done https://github.com/Mbed-TLS/mbedtls/pull/7641/commits/5fe32fff960cb100dfb3128f93c9e222c8709b79